compiler-core/src/ast/constant.rs

```diff
@@ -71,6 +73,34 @@ impl TypedConstant {
             | Constant::Var { typ, .. } => typ.clone(),
         }
     }
+
+    pub fn private_fn_deps(&self, already_found: &mut HashSet<EcoString>) {
+        match self {
+            Constant::Int { .. } => todo!(),
+            Constant::Float { .. } => todo!(),
+            Constant::String { .. } => todo!(),
+
+            TypedConstant::Var {
+                name, constructor, ..
+            } => {
+                if let Some(ValueConstructor { type_, .. }) = constructor.as_deref() {
+                    if let Type::Fn { .. } = **type_ {
+                        let _ = already_found.insert(name.clone());
+                    }
+                }
+            }
+
+            TypedConstant::Record { args, .. } => args
+                .iter()
+                .for_each(|arg| arg.value.private_fn_deps(already_found)),
+
+            Constant::Tuple { elements, .. } => todo!(),
+
+            Constant::List { elements, .. } => todo!(),
+
+            Constant::BitArray { segments, .. } => todo!(),
```

Comment:
I've added these missing cases which were not covered previously. It would be good to have tests for list and tuple (bit array isn't possible).